Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java

```diff
@@ -1004,42 +1062,107 @@ public RandomCutForest build() {
+            state.setCenterOfMassEnabled(forest.centerOfMassEnabled);
+            state.setOutputAfter(forest.outputAfter);
+            state.setParallelExecutionEnabled(forest.parallelExecutionEnabled);
+            state.setThreadPoolSize(forest.threadPoolSize);
+            state.setStoreSequenceIndexesEnabled(forest.storeSequenceIndexesEnabled);
+            state.setEntreesSeen(forest.getTotalUpdates());
+            state.setCompactEnabled(forest.compactEnabled);
+            state.setSaveTreeData(forest.saveTreeData);
+
+            if (forest.compactEnabled) {
+                Object coordinatorState = new PointStoreCoordinator.StateMapper()
+                        .toState((PointStoreCoordinator) forest.updateCoordinator);
+                state.setCoordinatorState(coordinatorState);
+
+                List<Object> componentModelStates = new ArrayList<>();
+                CompactSampler.StateMapper samplerMapper = new CompactSampler.StateMapper();
+                CompactRandomCutTreeDouble.StateMapper treeMapper = new CompactRandomCutTreeDouble.StateMapper();
+
+                for (IComponentModel<?> model : forest.componentModels) {
+                    SamplerPlusTree<Integer> samplerPlusTree = (SamplerPlusTree<Integer>) model;
+                    CompactSamplerPlusTreeState modelState = new CompactSamplerPlusTreeState();
+                    CompactSampler sampler = (CompactSampler) samplerPlusTree.getSampler();
+                    modelState.setSamplerState(samplerMapper.toState(sampler));
+                    if (forest.saveTreeData) {
+                        CompactRandomCutTreeDouble tree = (CompactRandomCutTreeDouble) samplerPlusTree.getTree();
+                        modelState.setTreeState(treeMapper.toState(tree));
+                    }
+                    componentModelStates.add(modelState);
+                }
+                state.setComponentModelStates(componentModelStates);
+            } else {
+                List<Object> componentModelStates = new ArrayList<>();
+                IContextualStateMapper<SimpleStreamSampler<double[]>, ?, ?> mapper;
+                if (forest.storeSequenceIndexesEnabled) {
+                    mapper = new SimpleStreamSampler.ArraySequentialStateMapper();
+                } else {
+                    mapper = new SimpleStreamSampler.ArrayStateMapper();
+                }
+
+                for (IComponentModel<?> model : forest.componentModels) {
+                    SamplerPlusTree<double[]> samplerPlusTree = (SamplerPlusTree<double[]>) model;
+                    SimpleStreamSampler<double[]> sampler = (SimpleStreamSampler<double[]>) samplerPlusTree
+                            .getSampler();
+                    componentModelStates.add(mapper.toState(sampler));
+                }
+            }
+
+            return state;
+        }
+
+        @Override
+        public RandomCutForest toModel(RandomCutForestState state) {
+            return null;
```

Comment:
Can we add a todo or maybe throw UnsupportedOperationException?

Reply:
Done.